src/runtime_src/tools/xclbinutil/SectionIPLayout.cxx

```diff
@@ -119,6 +119,75 @@ SectionIPLayout::getIPControlType(std::string& _sIPControlType) const
   throw std::runtime_error(errMsg);
 }
 
+const std::string
+SectionIPLayout::getFunctionalStr(PS_FUNCTIONAL eFunctional) const
+{
+  switch (eFunctional) {
+    case FC_DPU:
+      return "DPU";
+    case FC_PREPOST:
+      return "PrePost";
+    default:
+      return (boost::format("UNKNOWN (%d)") % static_cast<unsigned int>(eFunctional)).str();
+  }
+}
+
+PS_FUNCTIONAL
+SectionIPLayout::getFunctionalNoError(std::string& sFunctional)
+{
+  if (sFunctional == "DPU") 
+    return FC_DPU;
+  else if (sFunctional == "PrePost") 
+    return FC_PREPOST;
+  else
+    return FC_UNKNOWN;
```

Comment:
In this case, an exception should be thrown and caught by the calling method.

Reply:
code refactored